jwst/lib/set_telescope_pointing.py

```diff
@@ -196,6 +198,50 @@ def update_wcs(model, default_pa_v3=0., **kwargs):
         wcsaxes = 2
     model.meta.wcsinfo.wcsaxes = max(2, wcsaxes)
 
+    # Calculate S_REGION with the footprint
+    # information
+    update_s_region(model)
+
+
+def update_s_region(model):
+    """Update S_REGION sky footprint information
+
+    The S_REGION keyword is intended to store the spatial footprint of
+    an imaging observation using the VO standard STCS representation.
+
+    Parameters
+    ----------
+    model: jwst.datamodels.DataModel
+        The model to update in-place.
+    """
+
+    # Get the WCS info from the model
+    wcsinfo = model.meta.wcsinfo
+
+    # Using astropy.wcs, the most fullproof method
+    # of getting the wcs is to put everything in a FITS
+    # header and initialize from that.
+    header = Header(wcsinfo.instance)
+    wcs = WCS(header)
+    footprint = wcs.calc_footprint(undistort=True, center=True)
```

Comment:
It was suggested by Brian, so I left it in for consistency.  If it may be detrimental, I'll remove, but if the concept is, as stated in another issue/pr, to keep this updated all the way through, this routine could be the central one to use, maybe.